tools/xtask/src/tasks/license.rs

```diff
@@ -0,0 +1,80 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use anyhow::{anyhow, Result};
+use clap::Parser;
+
+use tracing::info;
+
+use cargo_metadata::Metadata;
+
+#[derive(Parser)]
+pub struct LicenseArgs {}
+
+pub fn run_license(_args: LicenseArgs, workspace: Metadata) -> Result<()> {
+    const LICENSE_FILES: &[&str] = &["LICENSE-MIT"];
+    let default_license = workspace.workspace_root.join(LICENSE_FILES[0]);
+
+    for package in workspace.workspace_packages() {
+        let crate_path = package
+            .manifest_path
+            .parent()
+            .ok_or_else(|| anyhow!("no crate directory for '{}'", package.name))?;
+
+        if let Some((symlink_path, license_index)) =
+            get_first_license_symlink_path(crate_path, LICENSE_FILES)
+        {
+            let root_license_path = pathdiff::diff_paths(
+                workspace.workspace_root.join(LICENSE_FILES[license_index]),
```

Comment:
If I understand everything correctly, the path to the license file in the workspace is calculated several times via `workspace.workspace_root.join(LICENSE_FILES[license_index])`. It is better to put this in a variable and reuse it.